pairinteraction/CMakeLists.txt

```diff
@@ -140,6 +140,48 @@ if (NOT MKL_FOUND AND PKG_CONFIG_FOUND)
   endif()
 endif()
 
+if(MSVC)
+  # https://www.intel.com/content/www/us/en/docs/onemkl/developer-guide-windows/2023-1/contents-of-the-redist-intel64-directory.html
+  file(GLOB MKL_DLLS
+    # Threading layer
+    "${MKL_ROOT}/redist/intel64/mkl_${MKL_THREADING}*.dll"
+    # Computational layer
+    "${MKL_ROOT}/redist/intel64/mkl_core*.dll"
+    "${MKL_ROOT}/redist/intel64/mkl_def*.dll"
+    "${MKL_ROOT}/redist/intel64/mkl_mc*.dll"
+    "${MKL_ROOT}/redist/intel64/mkl_avx*.dll"
+    "${MKL_ROOT}/redist/intel64/mkl_vml_def*.dll"
+    "${MKL_ROOT}/redist/intel64/mkl_vml_mc*.dll"
+    "${MKL_ROOT}/redist/intel64/mkl_vml_avx*.dll"
+    "${MKL_ROOT}/redist/intel64/mkl_vml_cmpt*.dll"
+    "${MKL_ROOT}/redist/intel64/libimalloc.dll")
```

Comment:
[`MKLConfig.cmake`](https://termbin.com/wh1x) (line 634) only looks for versioned DLLs. However, there are some issues with us also looking for only versioned DLLs.

1. MKL looks for only some versioned DLLs, e.g. `mkl_intel_thread.2.dll`, but this library in turn is linked against, say `mkl_avx2.3.dll`. We can't resolve this transitive versioning by only introspecting the targets exposed my `MKLConfig.cmake`.
2. I estimate the chances that there are both versioned and unversioned DLLs in the same redist folder as relatively slim, so extensively globbing should not pose a big problem.

This is what has been globbed for this particular installation of MKL:
```
libimalloc.dll
libiomp5md.dll
mkl_avx.2.dll
mkl_avx2.2.dll
mkl_avx512.2.dll
mkl_core.2.dll
mkl_def.2.dll
mkl_intel_thread.2.dll
mkl_mc.2.dll
mkl_mc3.2.dll
mkl_vml_avx.2.dll
mkl_vml_avx2.2.dll
mkl_vml_avx512.2.dll
mkl_vml_cmpt.2.dll
mkl_vml_def.2.dll
mkl_vml_mc.2.dll
mkl_vml_mc3.2.dll
```

Reply:
On top of that we could also define `MKL_DLLS` as a `CACHE` variable, so it can be overridden by a user on the command line.